src/ModelContextProtocol-SemanticKernel/Extensions/KernelExtensions.Vscode.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Stef Heyenrath
+
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel;
+using ModelContextProtocol.SemanticKernel.Types;
+using System.Text.Json;
+
+namespace ModelContextProtocol.SemanticKernel.Extensions;
+
+/// <summary>
+/// Extension methods for KernelPlugin
+/// </summary>
+public static partial class KernelExtensions
+{
+    /// <summary>
+    /// Adds Model Content Protocol plugins from the Visual Studio settings file (<c>settings.json</c>) and adds it into the plugin collection.
+    /// </summary>
+    /// <param name="plugins">The plugin collection to which the new plugin should be added.</param>
+    /// <param name="path">Path to the workspace settings file, for user settings use alternative overload</param>
+    /// <param name="inputs">Provide non-interactive values for inputs, VS Code would prompt user for it</param>
+    /// <param name="loggerFactory">The optional <see cref="ILoggerFactory"/>.</param>
+    /// <param name="cancellationToken">The optional <see cref="CancellationToken"/>.</param>
+    /// <returns><see cref="KernelPluginCollection"/> containing the functions provided in plugins.</returns>
+    public static async Task<KernelPluginCollection> AddToolsFromVsCodeConfigAsync(
+        this KernelPluginCollection plugins,
+        string path,
+        IEnumerable<VSCodeInput> inputs = null,
+        ILoggerFactory? loggerFactory = null,
+        CancellationToken cancellationToken = default)
+    {
+        var config = JsonSerializer.Deserialize<VsCodeSettings>(File.OpenRead(path));
```

Comment:
I think there are also Async versions from Json and File?

Reply:
OpenRead not - as it's using a stream. But I've made the Deserialize Async now.